packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmListInternal.kt

```diff
@@ -46,13 +46,21 @@ import kotlin.reflect.KClass
 /**
  * Implementation for unmanaged lists, backed by a [MutableList].
  */
-internal class UnmanagedRealmList<E> : RealmList<E>, InternalDeleteable, MutableList<E> by mutableListOf() {
+internal class UnmanagedRealmList<E>(
+    private val backingList: MutableList<E> = mutableListOf()
```

Comment:
Any specific reason for keeping a separate reference to this? You could also just implement `equals` and `hashCode` by delegating to `MutableList` with 

```
override fun equals(other: Any?): Boolean = super<MutableList>.equals(other)
override fun hashCode(): Int = super<MutableList>.hashCode()
```

Reply:
Because I didn't know that that expression was possible.